testsuite/MDAnalysisTests/coordinates/test_chainreader.py

```diff
@@ -123,34 +130,184 @@ class TestChainReaderCommonDt(object):
+            PSF, [DCD, CRD, DCD, CRD, DCD, CRD, CRD], dt=self.common_dt)
         return universe.trajectory
 
     def test_time(self, trajectory):
         # We test this for the beginning, middle and end of the trajectory.
         for frame_n in (0, trajectory.n_frames // 2, -1):
             trajectory[frame_n]
-            assert_almost_equal(trajectory.time,
-                                trajectory.frame * self.common_dt,
-                                5,
-                                err_msg="Wrong time for frame {0:d}".format(
-                                    frame_n))
+            assert_almost_equal(
+                trajectory.time,
+                trajectory.frame * self.common_dt,
+                5,
+                err_msg="Wrong time for frame {0:d}".format(frame_n))
 
 
 class TestChainReaderFormats(object):
     """Test of ChainReader with explicit formats (Issue 76)."""
 
     def test_set_all_format_tuples(self):
-        universe = mda.Universe(GRO, [(PDB, 'pdb'), (XTC, 'xtc'),
-                                      (TRR, 'trr')])
-        assert_equal(universe.trajectory.n_frames, 21)
+        universe = mda.Universe(GRO, [(PDB, 'pdb'), (XTC, 'xtc'), (TRR,
+                                                                   'trr')])
+        assert universe.trajectory.n_frames == 21
+        assert_equal(universe.trajectory.filenames, [PDB, XTC, TRR])
 
     def test_set_one_format_tuple(self):
         universe = mda.Universe(PSF, [(PDB_small, 'pdb'), DCD])
-        assert_equal(universe.trajectory.n_frames, 99)
+        assert universe.trajectory.n_frames == 99
 
     def test_set_all_formats(self):
         universe = mda.Universe(PSF, [PDB_small, PDB_closed], format='pdb')
-        assert_equal(universe.trajectory.n_frames, 2)
+        assert universe.trajectory.n_frames == 2
+
+
+# single frame doesn't work with xtc! NEED TO ACCOUND FOR THAT IN TESTING
+
+class TestChainReaderContinuous(object):
+    # description of the frame patterns to test are in comments as lists
+    @pytest.fixture
+    def top(self):
+        return atom_gro
+
+    @pytest.fixture
+    def trajs(self):
+        return [atom_0, atom_1, atom_2]
+
+    # [0 1 2 3 4 5 6 7 8 9] [0 1 2 4]
+    def test_last_frames(self, top):
+        # TODO check only one trajectory is used
+        u = mda.Universe(top, [allframes, atom_0], continuous=True)
+        assert u.trajectory.n_frames == 10
+        for i, ts in enumerate(u.trajectory):
+            assert_almost_equal(i, ts.time, decimal=4)
+
+    # [0 1 2 4] [0 1 2 3 4 5 6 7 8 9]
+    def test_last_traj_complete(self, top):
+        # TODO check only one trajectory is used
+        u = mda.Universe(top, [atom_0, allframes], continuous=True)
+        assert u.trajectory.n_frames == 10
+        for i, ts in enumerate(u.trajectory):
+            assert_almost_equal(i, ts.time, decimal=4)
+
+    # [0 1 2 3] [2] [3] [2 3 4 5 6] [5 6 7 8 9]
+    def test_middle_frames(self, top):
```

Comment:
So this is a bit tricky as it turns out. The sorting function correctly produces the order

- [0 1 2 3]
- [2]
- [2 3 4 5 6]
- [3]
- [5 6 7 8]


Now, this is because we sort by start times. The problem is that our current algorithm to find the start times doesn't see that some parts of the trajectory are completely included in another file.  In this case here the trajectory `[3]` in included in `[2 3 4 5]`. I say this reader should ideally be removed from the list. So this is the next task to tackle. 

@ianmkenney  help is appreciated. Getting all corner cases right isn't that easy.